cunumeric/deferred.py

```diff
@@ -1132,6 +1132,40 @@ def dot(self, src1, src2, stacklevel=0, callsite=None):
                 f"dot between {rhs1_array.ndim}d and {rhs2_array.ndim}d arrays"
             )
 
+    # Create array from input array and indices
+    @profile
+    @auto_convert([1])
+    @shadow_debug("choose", [1])
+    def choose(
+        self,
+        *args,
+        choices,
+        out,
+        is_tuple,
+        mode="rase",
+        stacklevel=0,
+        callsite=None,
+    ):
+        out_arr = out.base
+        index_arr = self._broadcast(out_arr.shape)
+        ch_tuple = tuple()
+        for c in args:
+            c = c._broadcast(out_arr.shape)
+            ch_tuple = ch_tuple + (c,)
+        task = self.context.create_task(CuNumericOpCode.CHOOSE)
+        task.add_output(out_arr)
+        task.add_input(index_arr)
+        if not is_tuple:
+            task.add_input(choices.base)
+            task.add_broadcast(choices.base)
+        else:
+            for c in ch_tuple:
+                task.add_input(c)
+                task.add_broadcast(c)
+        task.add_scalar_arg(is_tuple, bool)
+        task.add_alignment(out_arr, index_arr)
```

Comment:
If choices arrays are passed as separate arguments, their partitioning needs to also be aligned with the other arrays.